base/libgit2/utils.jl

```diff
@@ -1,6 +1,12 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
-const urlmatcher = r"^(http[s]?|git|ssh)?(:\/\/)?((\w+)@)?([A-Za-z0-9\-\.]+)(:[0-9]+)?(.*)$"
+const URL_REGEX = r"""
+^(?:(?<scheme>https?|git|ssh)\:\/\/)?
+(?:(?<user>.*?)(?:\:(?<password>.*?))?@)?
+(?<host>[A-Za-z0-9\-\.]+)
+(?:\:(?<port>\d+)?)?
+(?<path>.*?)$
+"""x
```

Comment:
Fixes several things with the original regex including:
- Doesn't fail or absorb password information into the username
- SSH URLs would have the colon included in the path capture. e.g. "git@github.com:JuliaLang/Example.jl" would have the path ":JuliaLang/Example.jl"
- Allows more complicated usernames including ones using hypens
- Switched to using named groups for clarity